clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/OpenstackAttributeValidator.groovy

```diff
@@ -102,6 +104,46 @@ class OpenstackAttributeValidator {
     result
   }
 
+  boolean validateHeatTemplate(String value, String attribute, AccountCredentialsProvider accountCredentialsProvider, String account) {
+    def result
+    def credentials = accountCredentialsProvider.getCredentials(account)
+    def client = ((OpenstackCredentials)credentials.getCredentials()).getProvider().getClient()
+    if (client && client.heat().templates().validateTemplate(value).isValid()) {
+      result = true
+    } else {
+      errors.rejectValue("${context}.${attribute}", "${context}.${attribute}.notValidHeatTemplate")
+      result = false
+    }
+    result
+  }
+
+  def validateApplication(String value, String attribute) {
+    if (validateNotEmpty(value, attribute)) {
+      return validateByRegex(value, attribute, prefixPattern)
+    } else {
+      errors.rejectValue("${context}.${attribute}", "${context}.${attribute}.invalid (Must match ${prefixPattern})")
```

Comment:
This will result in calling `errors.rejectValue` for in `validateNotEmpty` and here, correct? I don't know how big of a deal that is. Now I feel bad for possibly suggesting adding this `rejectValue` call here.